### ISSUE TITLE
Query Params can crash the site

### ISSUE THREAD

**Issue by beshaya** [microcovid/microcovid#317]
I found this one: if you get query params that don't specify a location, it causes PrevalanceControls to crash.

There may be others.

This is directly caused useQueryDataIfPresent erroneously passing variables set to `undefined` into the calculator:
```
export const useQueryDataIfPresent = (
  queryData: QueryData,
  calcData: CalculatorData,
): CalculatorData => {
  const queryDataFiltered = pickBy(queryData, (v) => v !== undefined)
  if (isEmpty(queryDataFiltered)) {
    return calcData
  } else {
    return { ...defaultValues, ...queryData }
  }
}
```

**Comment by blanchardjeremy** [comment]
Good catch.

Seems like we might also need to expand our test framework so we can have coverage for situations like this.

**Comment by beshaya** [comment]
To reproduce the bug in production:
* load a preset
* clear the topLocation
* refresh the page

Expected behavior: page loads
Actual behavior: page crashes

**Comment by beshaya** [comment]
More automated tests would be great. We don't have any tests for our React Components at the moment...

**Comment by blanchardjeremy** [comment]
Closed with #318
